solidity/contracts/bridge/Bridge.sol

```diff
@@ -1195,21 +1211,54 @@ contract Bridge is Ownable, EcdsaWalletOwner {
             abi.encodePacked(hex"160014", walletPubKeyHash)
         );
 
+        return
+            processRedemptionTxOutputs(
+                redemptionTxOutputVector,
+                walletPubKeyHash,
+                RedemptionTxOutputsProcessingInfo(
+                    outputStartingIndex,
+                    outputsCount,
+                    walletP2PKHScriptKeccak,
+                    walletP2WPKHScriptKeccak
+                )
+            );
+    }
+
+    /// @notice Processes all outputs from the redemption transaction. Tries to
+    ///         identify output as a change output, pending redemption request
+    //          or reported redemption. Reverts if one of the outputs cannot be
+    ///         recognized properly. Marks each request as processed by removing
+    ///         them from `pendingRedemptions` mapping.
+    /// @param redemptionTxOutputVector Bitcoin redemption transaction output
+    ///        vector. This function assumes vector's structure is valid so it
+    ///        must be validated using e.g. `BTCUtils.validateVout` function
+    ///        before it is passed here
+    /// @param walletPubKeyHash 20-byte public key hash (computed using Bitcoin
+    //         HASH160 over the compressed ECDSA public key) of the wallet which
+    ///        performed the redemption transaction.
+    /// @param processInfo RedemptionTxOutputsProcessingInfo identifying output
+    ///        starting index, the number of outputs and possible wallet change
+    ///        P2PKH and P2WPKH scripts.
+    function processRedemptionTxOutputs(
```

Comment:
Would it work if we don't extract this function and just stick with `processNonChangeRedemptionTxOutput`? I see some cons of having it here:
- It overloads the `processRedemptionTxOutputs` name so we have two functions named that way and the difference between them is very vague. On the other hand, it is quite difficult to name it differently.  This is the first sign indicating that this function should not be split.
- Usage of `RedemptionTxOutputsProcessingInfo` looks kinda weird as we could compute those parameters entirely within the body of this function. The point where we end with `processRedemptionTxOutputs` no.1 and start `processRedemptionTxOutputs` no.2 seems to be unintuitive. This is the second argument against splitting.

Reply:
> Would it work if we don't extract this function and just stick with processNonChangeRedemptionTxOutput? 

Unfortunately no. That was my first approach but the "stack too deep" error was still there. What's more, I think we will have to approach a refactoring of this function one more time once we add support to routing deposits to vaults. When optimizing gas costs in the PR I am working on, I tried to capture `self.bank` before using it to increase balances (the very end of the function) and got hit by "stack too deep" again.

> It overloads the processRedemptionTxOutputs name so we have two functions named that way and the difference between them is very vague. On the other hand, it is quite difficult to name it differently. This is the first sign indicating that this function should not be split.

I think it is fine if one function is preparing parameters for another. I used this pattern in the past (even outside of Solidity) and I think it can improve the clarity of the code and avoid too long functions. But sure, it's just a subjective opinion. 

> Usage of RedemptionTxOutputsProcessingInfo looks kinda weird as we could compute those parameters entirely within the body of this function. The point where we end with processRedemptionTxOutputs no.1 and start processRedemptionTxOutputs no.2 seems to be unintuitive. 

Agreed, I absolutely don't love it. I tried [various options](https://soliditydeveloper.com/stacktoodeep) and this one was working without a serious overhaul that I tried to avoid during this refactor. I completely agree it would be ideal to avoid it. I think we can try to approach it again when adding support for vaults. We may need to come up with another composition of logic because, as mentioned earlier, the "stack too deep" risk is still there.

Reply:
Ok, thanks for the clarification! If we can't do anything here, let's just leave it as is. Seems like an acceptable trade-off in the context of what you said.